### PR TITLE
export plugin data to injection

### PR DESCRIPTION
**Pull request by yoavkarako**

### CODE REVIEW

File: services/src/modules/arguments-injection/arguments-injection.ts [hunk 1]
```diff
@@ -25,14 +27,15 @@ function injectTemplate<T = unknown>(
     exports: info && context && getExportsProxy(context.exports, info?.parentType, source as Record<string, unknown>),
     vars: info?.variableValues,
     info,
+    pluginData: context?.resourceGroup?.pluginData,
```
Comment (AleF83) [inline services/src/modules/arguments-injection/arguments-injection.ts:30]: `pluginData` is a bit long.
```suggestion
    plugins: context?.resourceGroup?.pluginData,
```

File: services/tests/e2e/basic/plugins.schema.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ export const schema: Schema = {
         @policy(namespace: "plugins", name: "whole-rg-policy")
       pl_new_foo: Point! @localResolver(value: { x: 1, y: 2 })
       pl_new_bar: String! @localResolver(value: "NEW BAR") @reverse
+      pl_new_data: String! @localResolver(value: "{pluginData.foo}")
```
Comment (AleF83) [inline services/tests/e2e/basic/plugins.schema.ts:23]:
```suggestion
      pl_new_data: String! @localResolver(value: "{plugins.foo}")
```

File: services/tests/e2e/config/plugins/transform-resource-group.js [hunk 3]
```diff
@@ -24,6 +24,8 @@ module.exports = {
       newRg.policies.push(wholeRgPolicy);
     }
 
+    newRg.pluginData = { foo: 'bar' };
```
Comment (AleF83) [inline services/tests/e2e/config/plugins/transform-resource-group.js:27]: `pluginData` should be initialized to an empty object before applying plugins on resource group. So plugins will add some properties to the object rather then replace/create it.

**Review by AleF83 — CHANGES_REQUESTED** [review]
Run tests with `-u` flag to update snapshots. Check that the snapshots were changed as expected

**Comment by yoavkarako** [comment]
> Run tests with `-u` flag to update snapshots. Check that the snapshots were changed as expected

Can you do that? It won't run on my machine. I'll need to set it up the wsl...

**Comment by AleF83** [comment]
I've rebased a bit the branch and squashed the commits.
Just add some documentation [here](https://github.com/Soluto/stitch/blob/master/docs/arguments_injection.md) and [here](https://github.com/Soluto/stitch/blob/master/docs/plugins.md#transformresourcegroup)

**Review by AleF83 — APPROVED** [review]

**Pull request merged**